assemblies/assembly-planner-configuration.adoc

```diff
@@ -0,0 +1,1986 @@
++
+[source,xml,options="nowrap"]
+----
+<configuration>
+
+  <logger name="org.optaplanner" level="_LEVEL_"/>
+
+  ...
+
+</configuration>
+----
+. Optional: If you have a multitenant application where multiple `Solver` instances might be running at the same time, separate the logging of each instance into separate files:
+.. Surround the `solve()` call with  http://logback.qos.ch/manual/mdc.html[Mapped Diagnostic Context] (MDC):
++
+[source,java,options="nowrap"]
+----
+        MDC.put("tenant.name",tenantName);
+        MySolution bestSolution = solver.solve(problem);
+        MDC.remove("tenant.name");
+----
+
+.. Configure your logger to use different files for each ``${tenant.name}``.
+For example, use a `SiftingAppender` in the ``logback.xml`` file:
++
+[source,xml,options="nowrap"]
+----
+  <appender name="fileAppender" class="ch.qos.logback.classic.sift.SiftingAppender">
+    <discriminator>
+      <key>tenant.name</key>
+      <defaultValue>unknown</defaultValue>
+    </discriminator>
+    <sift>
+      <appender name="fileAppender.${tenant.name}" class="...FileAppender">
+        <file>local/log/optaplanner-${tenant.name}.log</file>
+        ...
+      </appender>
+    </sift>
+  </appender>
+----
++
+[NOTE]
+====
+When running multiple solvers or one multithreaded solve,
+most appenders, including the console, cause congestion with `debug` and `trace` logging.
+Switch to an async appender to avoid this problem or turn off `debug` logging.
+====
+
+. If OptaPlanner doesn't recognize the new level, temporarily add the system property `-Dlogback._LEVEL_=true` to troubleshoot.
+
+
+
+=== Using Log4J to log  OptaPlanner solver activity
+If you are already using Log4J and you do not want to switch to its faster successor, Logback, you can configure your OptaPlanner for Log4J.
+
+.Prerequisites
+* You have an OptaPlanner project
+* You are using the Log4J logging framework
+
+.Procedure
+
+. Add the bridge dependency to the project `pom.xml` file:
++
+[source,xml,options="nowrap"]
+----
+    <dependency>
+      <groupId>org.slf4j</groupId>
+      <artifactId>slf4j-log4j12</artifactId>
+      <version>1.x</version>
+    </dependency>
+----
+
+. Configure the logging level on the package `org.optaplanner` in your `log4j.xml` file as shown in the following example, where `_LEVEL_` is a logging level listed in XREF:
++
+[source,xml,options="nowrap"]
+----
+<log4j:configuration xmlns:log4j="http://jakarta.apache.org/log4j/">
+
+  <category name="org.optaplanner">
+    <priority value="_LEVEL_" />
+  </category>
+
+  ...
+
+</log4j:configuration>
+----
+
+. Optional: If you have a multitenant application where multiple `Solver` instances might be running at the same time, separate the logging of each instance into separate files:
++
+COMMENT: this section is about LOG4J so can we have a LOG4J example here?
```

Comment:
IDK what do you mean by the example? 
logging setup is provided by dependency and config file. 
Everything is logged by  SLF4J then if user mentioned dependency logback-classic then logback woudl be used in other case slf4j-log4j12 then log4j would be used.

The ``` MDC.put("tenant.name",tenantName);``` section about separation stays the same for log4j and logback

Reply:
OK thanks. That's what I meant.

Reply:
I put that section back into the Log4J procedure.